chemprop/cli/predict.py

```diff
@@ -278,7 +278,7 @@ def make_prediction_for_models(
         # TODO: might want to write a shared function for this as train.py might also want to do this.
         preds = torch.concat(predss, 0)
         if isinstance(model.predictor, MulticlassClassificationFFN):
-            preds = torch.argmax(preds, dim=-1)
+            preds = torch.argmax(preds, dim=2)
```

Comment:
```suggestion
            preds[..., 0] = torch.argmax(preds[..., 0], dim=2)
```
Making predictions in the shape of $(b,t,c,s)$ allows you to always use `preds = preds[..., 0]` no matter what predictors you are using. However, inevitably, you would still need to deal with `torch.argmax` for the multiclass classification outputs here. Although we only have $s=1$, this line of code is functionally correct in our codebase but not correct for $s>1$.

Reply:
Needing to desl with `torch.argmax` is true regardless of whether we use $(b, t, c, s)$ or $(b, t, c)$ though, so it's not really an argument to take $(b, t, c)$ over $(b, t, c, s)$.

I'm more referring to the following part of code, where we don't need to check and can always use `preds = preds[..., 0]` before using the metrics. Since `_evaluate_batch` is called very frequently, we don't want to need to use a if-else here to check what type of tasks we are handling.

```
    def _evaluate_batch(self, batch) -> list[Tensor]:
        bmg, V_d, X_d, targets, _, lt_mask, gt_mask = batch

        mask = targets.isfinite()
        targets = targets.nan_to_num(nan=0.0)
        preds = self(bmg, V_d, X_d)
        weights = torch.ones_like(targets)

        preds = preds[..., 0]

        return [
            metric(preds, targets, mask, weights, lt_mask, gt_mask) for metric in self.metrics[:-1]
        ]
```